tools/crm_node.c

```diff
@@ -573,11 +581,18 @@ purge_node_from_cib(const char *node_name, long node_id)
     if (node_id > 0) {
         crm_xml_set_id(xml, "%ld", node_id);
     }
-    rc = cib->cmds->remove(cib, XML_CIB_TAG_STATUS, xml, cib_sync_call);
+    rc = cib->cmds->remove(cib, XML_CIB_TAG_STATUS, xml,
+                           cib_sync_call|cib_transaction);
     rc = pcmk_legacy2rc(rc);
-    if (rc == pcmk_rc_ok) {
+
+    // Commit the transaction
+    rc = cib->cmds->end_transaction(cib, true, cib_scope_local);
```

Comment:
`cib_scope_local` only gets processed at all when the CIB manager is in legacy mode. It basically tells us to process locally instead of forwarding to the primary.

I'm afraid to remove the old uses of `cib_scope_local`, but I'm not sure that we need new uses. In particular, the current individual `remove` requests don't have `cib_scope_local`, so they get forwarded to the primary. I'd keep that behavior for the transaction unless you have a reason to deviate.

Reply:
However, we *do*` want to use `cib_sync_call` here to preserve existing behavior. If you want to make it asynchronous (the default for transaction commits), then we need a callback to check for success.